PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/SpringCloudDependencyManager.java

```diff
@@ -0,0 +1,145 @@
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.azure.springcloud.dependency.action;
+
+import com.microsoft.azure.common.exceptions.AzureExecutionException;
+import com.microsoft.intellij.util.PomXmlUpdater;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.dom4j.Document;
+import org.dom4j.DocumentException;
+import org.dom4j.DocumentFactory;
+import org.dom4j.DocumentHelper;
+import org.dom4j.Element;
+import org.dom4j.Node;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SpringCloudDependencyManager {
+    public static final String POM_NAMESPACE = "http://maven.apache.org/POM/4.0.0";
+    private Document doc;
+
+    public SpringCloudDependencyManager(String effectivePomXml) throws DocumentException {
+        Map<String, String> nsContext = new HashMap<>();
+        nsContext.put("ns", POM_NAMESPACE);
+        DocumentFactory.getInstance().setXPathNamespaceURIs(nsContext);
+        doc = DocumentHelper.parseText(effectivePomXml);
+    }
+
+    public Map<String, String> getCurrentVersions(String groupId) {
+        Map<String, String> res = new HashMap<>();
+        List<Node> nodes = doc.selectNodes("//ns:project/ns:dependencies/ns:dependency");
+        for (Node node : nodes) {
+            String thisGroupId = ((Element) node).elementTextTrim("groupId");
+            if (StringUtils.equals(thisGroupId, groupId)) {
+                res.put(((Element) node).elementTextTrim("artifactId"), ((Element) node).elementTextTrim("version"));
+            }
+        }
+
+        nodes = doc.selectNodes("//ns:project/ns:dependencyManagement/ns:dependencies/ns:dependency");
```

Comment:
duplicate with previous codes, we could get all the nodes first and then get the dependency version